packages/rpc/src/RpcMiddleware.ts

```diff
@@ -35,6 +35,19 @@ export interface RpcMiddleware<Provides, E> {
   }): Effect.Effect<Provides, E>
 }
 
+/**
+ * @since 1.0.0
+ * @category models
+ */
+export interface RpcMiddlewareWrap {
```

Comment:
@tim-smart Opted out for a completely different interface, would you prefer a typelevel parameter into RpcMiddleware instead and perform conditional arguments over there instead?